src/Tizen.Sensor/Tizen.Sensor/SensorRecorderDataManager.cs

```diff
@@ -0,0 +1,576 @@
+        /// </summary>
+        /// <returns>
+        /// returns the start and the end time of the time period of a given record data.
+        /// </returns>
+        /// <since_tizen> 6 </since_tizen>
+        public RecordTimePeriod DataGetTime()
+        {
+            long startTime = -1, endTime = -1;
+            if (CheckDataHandle())
+            {
+                int error = Interop.SensorRecoder.DataGetTime(dataHandle, out startTime, out endTime);
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error getting the sensor recorder time");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorderDataManager.DataGetTime Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Recorder data is not created");
+            }
+
+            return new RecordTimePeriod(startTime, endTime);
+        }
+
+        /// <summary>
+        ///  Queries the recorded data synchronously and updates the data handle
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public void Update(Query query)
+        {
+            _sensorRecorderDataCallBack = (SensorType _type, IntPtr _dataHandle, int remains, int _error, Int64 _userData) =>
+            {
+                if (_error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error in Update synchronously CallBack");
+                }
+                else
+                {
+                    dataHandle = _dataHandle;
+                }
+
+                Log.Debug(Globals.LogTag, "Recorder data callback recieved");
+                return true;
+            };
+
+            int error = Interop.SensorRecoder.Read(sensorType, query.queryHandle, _sensorRecorderDataCallBack, IntPtr.Zero);
+
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error in synchronously updating data handle");
+                throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorderDataManager.update Failed");
+            }
+        }
+
+        private void AsyncUpdater(IntPtr QueryHandle, Interop.SensorRecoder.SensorRecorderDataCb Cb, IntPtr _userData)
+        {
+            int error = Interop.SensorRecoder.ReadAsync(sensorType, QueryHandle, Cb, IntPtr.Zero);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error in asynchronously updating data handle");
+                throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorderDataManager.AsyncUpdater Failed");
+            }
+        }
+
+        /// <summary>
+        /// Queries the recorded data asynchronously and updates the data handle
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public async Task UpdateAsync(Query query)
```

Comment:
Parameters of methods must have valid documentation. Method has 1 parameter, 0 are documented.